### PR TITLE
fix(validation): fix url port validation

### PR DESCRIPTION
**Pull request by vladikoff**

### CODE REVIEW

File: lib/routes/validators.js [hunk 1]
```diff
@@ -170,9 +170,21 @@ function isValidUrl(url, hostnameRegex) {
...
+
+  // if the given url has a port :80 or :443 the `new URL` above will drop `parsed.port` for :80 and for :443
+  // to validate the url below we reconstruct that port
+  let forcedPort = ':443'
+  if (parsed.protocol === 'http') {
+    forcedPort = ':80'
+  }
+
+  if (
+    (parsed.origin + forcedPort + parsed.pathname !== url && parsed.origin + parsed.pathname !== url)
```
Comment (vladikoff) [inline lib/routes/validators.js]: this doesn't seem right, need to clean it up i think

File: lib/routes/validators.js [hunk 2]
```diff
@@ -170,6 +170,13 @@ function isValidUrl(url, hostnameRegex) {
   // The only trick here is that `new URL()` will add a trailing
   // slash if there's no path component, which is why we also
   // compare to `origin` below.
+
+  // if the original url has a :443 in the url we need to strip it for the check below
+  // to follow the `new URL` behaviour
+  if (/^https:\/\//.test(url) && /:443/.test(url)) {
```
Comment (shane-tomlinson) [inline lib/routes/validators.js]: I was testing this and think it needs to be:

```js
if (/^https:\/\//.test(url) && /:443($|\/[^\d])/.test(url)) {
```

That avoids `:4430` from having it's port replaced. It says, replace `:443 if it's at the end of the URL, or if the character following `:443` is not a digit

Comment (vladikoff) [reply lib/routes/validators.js]: that would be a good improvement 👍

Comment (jrgm) [reply lib/routes/validators.js]: Is there a reason not to use `url.parse` instead of regular expressions?

Comment (jrgm) [reply lib/routes/validators.js]: Actually, widening the view, `parsed.port` is already available (`null` if no explicit port number).

Comment (vladikoff) [reply lib/routes/validators.js]: @jrgm we already use `new URL` above

**Comment by vladikoff** [comment]
continued in https://github.com/mozilla/fxa-auth-server/pull/2616

**Pull request closed without merging**